plugins/ipam/host-local/config.go

```diff
@@ -39,31 +39,29 @@ type IPAMArgs struct {
 }
 
 type Net struct {
-	Name string      `json:"name"`
-	IPAM *IPAMConfig `json:"ipam"`
+	Name  string      `json:"name"`
+	IPAM  *IPAMConfig `json:"ipam"`
+	IPAM6 *IPAMConfig `json:"ipam6"`
 }
 
 // NewIPAMConfig creates a NetworkConfig from the given network name.
-func LoadIPAMConfig(bytes []byte, args string) (*IPAMConfig, error) {
+func LoadIPAMConfig(bytes []byte, args string) (*IPAMConfig, *IPAMConfig, error) {
 	n := Net{}
 	if err := json.Unmarshal(bytes, &n); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	if args != "" {
 		ipamArgs := IPAMArgs{}
 		err := types.LoadArgs(args, &ipamArgs)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 	}
 
-	if n.IPAM == nil {
-		return nil, fmt.Errorf("%q missing 'ipam' key")
+	if n.IPAM == nil && n.IPAM6 == nil {
+		return nil, nil, fmt.Errorf("Need at least one of ipam or ipam6 key")
 	}
 
-	// Copy net name into IPAM so not to drag Net struct around
```

Comment:
There should probably be some minimal validation here too, to ensure that IPv4 values aren't passed in 'ipam6' and vice versa.  Since NewIPAllocator() isn't passed AF_INET/AF_INET6 it has no idea what kind of config it's getting in the v4 and v6 blocks in main.go and it will happily accept IPv4 values in the IPv6 allocator, I think.